bionic/cache.py

```diff
@@ -317,7 +318,14 @@ def _file_from_value(self, value):
         value_path = dir_path / value_filename
 
         ensure_parent_dir_exists(value_path)
-        self.query.protocol.write(value, value_path)
+        try:
+            self.query.protocol.write(value, value_path)
+        except Exception as e:
+            raise EntitySerializationError(oneline(
+                f'''
+                Value of entity {self.query.entity_name!r}
+                could not be serialized to disk
+                ''')) from e
```

Comment:
what does `from e` do here?

Reply:
It creates a "chained exception" which includes both the current stack trace and `e`'s original stack trace. It's analogous to exception wrapping in Java:

    } catch (Exception e) {
      throw new OtherException("message", e);
    }

Reply:
this is neat!